18-explaining-models-and-predictions.Rmd

```diff
@@ -1,6 +1,313 @@
+
+Using `r pkg(DALEX)`, we compute this kind of variable importance via the `model_parts()` function.
+
+```{r}
+vip_lm <- model_parts(explainer_lm, loss_function = loss_root_mean_square) 
+vip_rf <- model_parts(explainer_rf, loss_function = loss_root_mean_square)
+```
+
+Again, we could use the default plot method from `r pkg(DALEX)` by calling `plot(vip_lm, vip_rf)` but the underlying data is available for exploration, analysis, and plotting.
+
+```{r}
+full_vip <- bind_rows(vip_lm, vip_rf) %>%
+  filter(variable != "_baseline_")
+
+full_rmse <- full_vip %>% 
+  filter(variable == "_full_model_") %>% 
+  group_by(label) %>% 
+  summarise(dropout_loss = mean(dropout_loss))
+
+full_vip %>%
+  filter(variable != "_full_model_") %>% 
+  mutate(variable = forcats::fct_reorder(variable, dropout_loss)) %>%
+  ggplot(aes(dropout_loss, variable, color = label, fill = label)) +
+  geom_vline(data = full_rmse, aes(xintercept = dropout_loss, color = label),
+             size = 1.4, lty = 2, alpha = 0.7) +
+  geom_boxplot(alpha = 0.2) +
+  facet_wrap(~label, ncol = 1) +
+  theme(legend.position = "none") +
+  labs(x = "RMSE after permutations (higher indicates more important)", y = NULL, fill = NULL, color = NULL)
+```
+
+The dashed line in each panel shows the RMSE for the full model, either the linear model or the random forest model. Features further to the right are more important, because permuting them results in higher RMSE. There is quite a lot of interesting information to learn from this plot; for example, neighborhood is quite important in the linear model with interactions/splines but the second least important feature for the random forest model.
+
+## Building global explanations from local explanations
+
+So far in this chapter, we have focused on local model explanations for a single observation (via Shapley additive explanations) and global model explanations for a data set as a whole (via permuting features). It is also possible to build global model explanations up by aggregating local model explanations, as with _partial dependence profiles_.
+
+
+:::rmdnote
+Partial dependence profiles show how the expected value of a model prediction, like the predicted price of a home in Ames, changes as a function of a feature, like the age or gross living area.
+:::
+
+One way to build such a profile is by aggregating or averaging profiles for individual observations. A profile showing how an individual observation's prediction changes as a function of a given feature is called an ICE (individual conditional expectation) profile or a CP (*ceteris paribus*) profile. We can compute such individual profiles (for 500 of the observations in our training set) and then aggregate them using the `r pkg(DALEX)` function `model_profile()`.
+
+```{r}
+pdp_age <- model_profile(explainer_rf, N = 500, variables = "Year_Built")
+```
+
```

Comment:
This kinds of makes it look like `model_profile()~ produces the plot. Maybe show the plot code?

Reply:
Since we use very similar code a few times, I'd make a function for this.